packages/eas-cli/src/project/android/applicationId.ts

```diff
@@ -33,6 +33,57 @@ export async function ensureApplicationIdIsDefinedForManagedProjectAsync(
   }
 }
 
+export class AmbiguousApplicationIdError extends Error {
+  constructor(message?: string) {
+    super(message ?? 'Could not resolve applicationId.');
+  }
+}
+
+export async function getApplicationIdFromBareAsync(
+  projectDir: string,
+  gradleContext?: GradleBuildContext
+): Promise<string> {
+  const errorMessage = 'Could not read application id from Android project.';
+
+  if (gradleContext) {
+    const buildGradle = await gradleUtils.getAppBuildGradleAsync(projectDir);
+    const applicationIdSuffix = gradleUtils.resolveConfigValue(
+      buildGradle,
+      'applicationIdSuffix',
+      gradleContext.flavor
+    );
+    if (applicationIdSuffix) {
+      throw new Error('applicationIdSuffix is not supported');
```

Comment:
1. I don't understand what this error message means.
2. So if I don't understand then I think that most of the users (if not all) won't understand it as well.